guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,927 @@
+#[table_name="users"]
+pub struct UpdateUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: Option<Option<&'a str>>,
+ }
+
+// Setting up the association to users
+#[derive(Identifiable, Associations, Queryable)]
+#[belongs_to(User)]
+pub struct Post {
+    pub id: i32,
+    pub user_id: i32,
+    pub title: String,
+    pub content: String,
+}
+
+// Lets us insert new posts
+#[derive(Insertable)]
+#[table_name="posts"]
+pub struct NewPost<'a> {
+    pub user_id: i32,
+    pub title: &'a str,
+    pub content: &'a str,
+ }
+```
+
+```rust
+// File: src/main.rs
+
+use diesel::prelude::*;
+
+fn main() {
+  // Creating and inserting new user
+  let new_user = NewUser { 
+      first_name: "Issac", 
+      last_name: "Kleiner", 
+      email: Some("issac.kleiner@blackmesa.co"),
+  };
+
+  let issac_kleiner = diesel::insert_into(users::table)
+      .values(&new_user)
+      .get_result::<User>(&db_connection)
+      .expect("Error inserting row into database");
+
+  // Setting up our posts vec and pointing each post to the 
+  // most recently inserted user
+  let post_list = vec![
+      NewPost {
+          user_id: issac_kleiner.id().to_owned(),
+          title: "Top Secret #001",
+          content: "I'm feeling optimistic about our new reactor code written in Rust!",
+      },
+      NewPost {
+          user_id: issac_kleiner.id().to_owned(),
+          title: "Top Secret #002",
+          content: "Finished making special pet for Gordon. I hope he likes it!",
+      }
+  ];
+
+  // Insert the new posts vector
+  diesel::insert_into(posts::table)
+      .values(&post_list)
+      .execute(&db_connection)
+      .expect("Error inserting post");
+
+  // Get the first user
+  let issac = users.first::<User>(&db_connection)
+      .expect("Couldn't find first user");
+
+  // Get all the posts belonging to the first user
+  let issacs_posts = Post::belonging_to(&issac)
+      .get_results::<Post>(&db_connection)
+      .expect("Couldn't find associated posts");
+
+  for post in issacs_posts {
+      println!("-----\nTitle: {}\nContent: {}\n", post.title, post.body);
+  }
+
+  // Outputs
+  /*
+------
+Title: Top Secret #001
+Content: "I'm feeling optimistic about our new reactor code written in Rust!"
+
+------
+Title: Top Secret #002
+Content: "Finished making special pet for Gordon. I hope he likes it!"
+
+  */
+}
+```
+
+If we were to forget to derive `Associations` on `Posts`,
+we would see the following error.
+
+```rust
+error[E0599]: no function or associated item named `belonging_to` found for type `diesel_demo::models::Post` in the current scope
+   --> src/bin/main.rs:103:24
+    |
+103 |     let issacs_posts = Post::belonging_to(&issac)
+    |                        ^^^^^^^^^^^^^^^^^^ function or associated item not found in `diesel_demo::models::Post`
+```
+
+## Conclusion
+
+Please check out other [official guides], [docs], and [gitter channel]
+for more information on using the Diesel framework.
```

Comment:
One of these is not like the other… I'd move the gitter channel to a separate sentence:

> Please check out other [official guides] and [API docs] for more information on using the Diesel framework.
> If you have any questions, join our [gitter channel], the Diesel team is happy to help.